### ISSUE TITLE
Subtitle settings will always show the first subtitle language

### ISSUE THREAD

**Issue by Sopor** [xbmc/xbmc#17580]
### Describe the bug
Here is a clear and concise description of what the problem is:

I have preferred language set to Swedish for subtitles. When i play a DVD ISO or IMG file i can choose language in the menu. So if the DVD has four languages to choose from (Danish, Finnish, Norwegian and Swedish) and i open the subtitle menu in Kodi it will still show it is on language 1, in this case Danish. The subtitle that is showing in the movie is Swedish.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

The subtitle settings should show the selected language.

## Actual Behavior

It will always show language 1, in this case Danish.

### To Reproduce

1. Play a DVD with more than one subtitles.
2. Change language in Kodi from the subtitle settings to any subtitle except the first one.
3. Close language settings and open it again and it will continue to show language 1.

### Debuglog

The debuglog can be found here:
https://paste.kodi.tv/ditevaxabu.kodi

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

This could have something to do with issue #17573 but i'm no expert. I run Kodi in English and i have `Preferred subtitle language` set to Swedish.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [x] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 10 / Vero 4K+ 2020.01-1
 - Kodi version: 18.6

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by leo06** [comment]
Did you try with GUI language also in swedish?

**Comment by Sopor** [comment]
@leo06 Yes and it will still show the first language both as subtitle in the movie and as selected subtitle in the subtitle settings. In Kodi 17.6 it works perfect.

**Comment by Sopor** [comment]
Update: These are DVD in STORED multi-archive RAR format. Maybe this is another vfs.rar issue?

**Comment by DaVukovic** [comment]
Seeing that as well. So I can confirm this issue

**Comment by Sopor** [comment]
Something else that i noticed is that even if subtitles are disabled the selected subtitle is showing in the `Now playing` window. In Kodi 17.6 it was not visible until i enable subtitles. IMO it should be hidden until the subtitles get enabled or else it will easy confuse the user that the subtitles are enabled.

**Comment by Sopor** [comment]
Tested in Kodi 18.6 Git:20200505-f0bc4746c8 and it is still not showing the right subtitle language
